[CI Flaky test]: test_chat_streaming_audio — streaming and non-streaming responses diverge at temperature=0

### Name of failing test

`entrypoints/multimodal/openai/chat_completion/test_audio.py::test_chat_streaming_audio[https://vllm-public-assets.s3.us-west-2.amazonaws.com/multimodal_asset/mary_had_lamb.ogg-fixie-ai/ultravox-v0_5-llama-3_2-1b]`

### Basic information

- [x] Flaky test
- [ ] Can reproduce locally
- [ ] Caused by external libraries (e.g. bug in `transformers`)

I have no GPU available, so I have not reproduced this locally — everything below is from Buildkite logs.

### 🧪 Describe the failing test

All file references are against `main` @ `b88916617d`.

Buildkite step `Entrypoints Integration (Multimodal)` fails in `pytest entrypoints/multimodal`, one test out of the 123 collected. `test_chat_streaming_audio` issues the same audio chat request twice against one server — once non-streaming, once streaming — both with `temperature=0.0` and `max_completion_tokens=8` (`tests/entrypoints/multimodal/openai/chat_completion/test_audio.py:265-304`), then asserts the two agree at `:304`:

```python
assert "".join(chunks) == output
```

The two responses disagree from the second token onwards:

```text
FAILED entrypoints/multimodal/openai/chat_completion/test_audio.py::test_chat_streaming_audio[...ultravox-v0_5-llama-3_2-1b] - AssertionError: assert 'A possible s...s audio could' == 'A short titl...udio could be'

  - A short title for this audio could be
  ?                                   ---
  + A possible short title for this audio could
  ?  +++++++++
======================================== 1 failed, 120 passed, 1 skipped, 1 xfailed, 16 warnings in 1663.69s (0:27:43) =========================================
```

Non-streaming returned `A short title for this audio could be`; streaming returned `A possible short title for this audio could`. The streaming response contains a word (`possible`) absent from the non-streaming one, so this is a divergence in the sampled token sequence at position 2, not a detokenization or chunk-boundary artifact. The earlier assertion at `:302` (`chunk.choices[0].finish_reason == stop_reason`) passed, so both responses stopped for the same reason — consistent with both hitting the 8-token cap.

Worth noting when judging how much numeric slack to expect here: the server fixture for this file runs `--dtype float32 --enforce-eager --max-num-seqs 5 --max-model-len 2048` (`:29-41`). Prefix caching is left at its default of enabled (`vllm/config/cache.py:93`), so the second request is a candidate to prefill against blocks the first one populated. I have not confirmed from the logs that it actually hit the cache, but it is the most obvious asymmetry between two otherwise identical greedy requests.

So the question I cannot answer without reproducing: is two identical greedy requests to one server disagreeing on token 2 — under fp32 and `--enforce-eager` — acceptable run-to-run variance for this configuration, or an invariance bug that this config should have been hiding?

### 📝 History of failing test

Observed runs of this step, ordered by build number:

| Build | PR | Result | Job finished (UTC) |
|---|---|---|---|
| [81189](https://buildkite.com/vllm/ci/builds/81189#019fb081-bd82-461a-a106-a0cf11a6b6f4) | #49660 | pass | 2026-07-30 01:42 |
| [81196](https://buildkite.com/vllm/ci/builds/81196#019fb0b8-7530-4e57-8ca1-a96a6a7d3441) | #48408 | **fail** | 2026-07-30 02:29 |
| [81198](https://buildkite.com/vllm/ci/builds/81198#019fb112-7879-44ab-8702-38a57df09e32) | #50352 | **fail** | — |
| [81199](https://buildkite.com/vllm/ci/builds/81199#019fb0c3-596c-4485-bea8-0df6ca6c8dc0) | #50305 | **fail** | 2026-07-30 02:42 |
| [81201](https://buildkite.com/vllm/ci/builds/81201#019fb0d4-c8da-4800-bcc0-17e0ac5f05d0) | #50357 | **fail** | 2026-07-30 03:02 |
| [81206](https://buildkite.com/vllm/ci/builds/81206#019fb11b-fa42-41be-9ef6-24df5f3033e2) | #50349 | **fail** | — |
| [81208](https://buildkite.com/vllm/ci/builds/81208#019fb0ea-9337-4819-8eed-b48fc911175d) | #50339 | pass | 2026-07-30 03:37 |
| [81214](https://buildkite.com/vllm/ci/builds/81214#019fb117-4c47-418b-ba44-c29c89f7650f) | #48408 | **fail** | 2026-07-30 04:14 |

Six failures in the seven runs after 81189. All six are byte-identical — same two strings, same diff — but 81208 passing in the middle of them is why I have labelled this flaky rather than broken: what is stable is the failure *signature*, not whether it fails. That fits token 2 being a near-tie with only two plausible continuations, so a coin-flip yields a repeatable-looking diff.

The failures span unrelated PRs, two of which cannot plausibly be the cause: #50357 only touches `docker/Dockerfile`, `docs/getting_started/installation/gpu.cuda.inc.md` and `.buildkite/scripts/hardware_ci/run-gh200-test.sh`, and #48408 only touches `vllm/v1/kv_offload/` and `vllm/distributed/kv_transfer/` files, with no KV connector configured in this test. All eight runs above were on H200 agents, each a different instance, so the agent hardware class is not the variable.

**No commit correlation found.** I tried to bisect this from the commits each log reports checking out (`HEAD is now at ...`), comparing them with `gh api repos/vllm-project/vllm/compare/<commit>...<checked-out commit>`. That line of investigation came up empty: the passing build 81208 checked out `274bcbd682`, from which every candidate I had narrowed to is reachable — #47301, #50297, #50326 and #47124 — as are the earlier changes that looked most likely to move greedy output (#49750 `[Perf] RMSNorm uncontiguous support`, #50244, #49660). So I have no evidence pointing at any particular `main` commit, and I would rather report that than guess.

For blast radius: this step's `source_file_dependencies` is `vllm/` (`.buildkite/test_areas/entrypoints.yaml:154-156`), so it runs for essentially any PR touching engine code, and at a 6-in-7 rate it will block most of them until this is understood.

I did not open a fix PR. The assertion is checking a real invariant, so silencing it with a retry marker or a looser comparison would hide whatever this is rather than fix it — but if maintainers judge that greedy output is not expected to be reproducible across two requests here, then the test itself is the thing that needs changing, and I am happy to do that instead.

**Duplicate check.** `gh issue list --repo vllm-project/vllm --state all --search "test_chat_streaming_audio"` returns only the closed #30854, and no open issue under the `ci-failure` label covers this test or step.

<details>
<summary>Log excerpt (build 81214, step <code>entrypoints-integration-multimodal</code>):</summary>

```text
=================================================================== short test summary info ====================================================================
FAILED entrypoints/multimodal/openai/chat_completion/test_audio.py::test_chat_streaming_audio[https://vllm-public-assets.s3.us-west-2.amazonaws.com/multimodal_asset/mary_had_lamb.ogg-fixie-ai/ultravox-v0_5-llama-3_2-1b] - AssertionError: assert 'A possible s...s audio could' == 'A short titl...udio could be'

  - A short title for this audio could be
  ?                                   ---
  + A possible short title for this audio could
  ?  +++++++++
======================================== 1 failed, 120 passed, 1 skipped, 1 xfailed, 16 warnings in 1663.69s (0:27:43) =========================================
🚨 Error: The command exited with status 1
```

</details>

### CC List.

@DarkLight1337 @NickLucche @ywang96 @robertgshaw2-redhat @aarnphm

--- a/tests/entrypoints/multimodal/openai/chat_completion/test_audio.py
+++ b/tests/entrypoints/multimodal/openai/chat_completion/test_audio.py
@@ -10,7 +10,6 @@ import pytest_asyncio
 from tests.utils import RemoteOpenAIServer
 from vllm.assets.audio import AudioAsset
 from vllm.multimodal.utils import encode_audio_base64, encode_audio_url, fetch_audio
-from vllm.platforms import current_platform
 
 MODEL_NAME = "fixie-ai/ultravox-v0_5-llama-3_2-1b"
 TEST_AUDIO_URLS = [
@@ -18,10 +17,6 @@ TEST_AUDIO_URLS = [
     AudioAsset("mary_had_lamb").url,
 ]
 MAXIMUM_AUDIOS = 2
-
-# Disable prefix caching on ROCm to reduce non-determinism in
-# streaming-vs-non-streaming comparisons.
-_ROCM_ARGS = ["--no-enable-prefix-caching"] if current_platform.is_rocm() else []
 
 
 @pytest.fixture(scope="module")
@@ -37,7 +32,7 @@ def server():
         "--trust-remote-code",
         "--limit-mm-per-prompt",
         json.dumps({"audio": MAXIMUM_AUDIOS}),
-        *_ROCM_ARGS,
+        "--no-enable-prefix-caching",
     ]
 
     with RemoteOpenAIServer(MODEL_NAME, args) as remote_server:

